### PR TITLE
Add a bunch of repoquery options

### PR DESCRIPTION
**Pull request by kontura**
Add several repoquery options.
For: https://github.com/rpm-software-management/dnf5/issues/122

### CODE REVIEW

File: dnf5/commands/repoquery/repoquery.cpp [hunk 1]
```diff
@@ -256,6 +256,16 @@ void RepoqueryCommand::set_argument_parser() {
         "Limit to unneeded installed packages (i.e. packages that were installed as "
         "dependencies but are no longer needed).",
         false);
+    extras = std::make_unique<libdnf::cli::session::BoolOption>(
+        *this, "extras", '\0', "Limit to packages that are not present in any of of available repositories.", false);
```
Comment (m-blaha) [inline dnf5/commands/repoquery/repoquery.cpp:264]: Maybe we should be more explicit here and state that the packages are **installed** - `Limit to installed packages...`. Does it make the description too long?
And there is a duplicated **of** - `...in any of of available...`

File: dnf5/commands/repoquery/repoquery.cpp [hunk 2]
```diff
@@ -256,6 +256,16 @@ void RepoqueryCommand::set_argument_parser() {
...
+    upgrades = std::make_unique<libdnf::cli::session::BoolOption>(
+        *this,
+        "upgrades",
+        '\0',
+        "Limit to packages that provide an upgrade for some already installed package.",
+        false);
+    recent = std::make_unique<libdnf::cli::session::BoolOption>(
+        *this, "recent", '\0', "Limit to only recently edited packages.", false);
```
Comment (m-blaha) [inline dnf5/commands/repoquery/repoquery.cpp:272]: I know it's also in dnf4 docs, but I consider word `edited` a bit odd in this context. Wouldn't be `changed` or `added` a better fit?

File: dnf5/commands/repoquery/repoquery.cpp (hunk 1 above)
Comment (kontura) [reply dnf5/commands/repoquery/repoquery.cpp]: I like it, I have also rephrased the ending a bit.

File: dnf5/commands/repoquery/repoquery.cpp (hunk 2 above)
Comment (kontura) [reply dnf5/commands/repoquery/repoquery.cpp]: `added` seems a bit misleading to me but `changed` is better, thanks. :+1:

**Comment by kontura** [comment]
> Please, add the new options also to `doc/commands/repoquery.8.rst` documentation.

Right, all of the new options are missing the docs I was thinking I will add it all at once in a separate PR if that is ok??
(There is also a task for that in https://github.com/rpm-software-management/dnf5/issues/122#issue-1437992549)

**Pull request merged**